packaging/debian-sid/patches/disable-i18n.patch

```diff
@@ -0,0 +1,289 @@
+-
+-	bindTextDomain("snappy-test", localeDir)
+-	os.Setenv("LANG", "en_DK.UTF-8")
+-	setLocale("")
+-}
+-
+-func (s *i18nTestSuite) TearDownTest(c *C) {
+-	os.Setenv("LANG", s.origLang)
+-	os.Setenv("LC_MESSAGES", s.origLcMessages)
+-}
+-
+-func (s *i18nTestSuite) TestTranslatedSingular(c *C) {
+-	// no G() to avoid adding the test string to snappy-pot
+-	var Gtest = G
+-	c.Assert(Gtest("singular"), Equals, "translated singular")
+-}
+-
+-func (s *i18nTestSuite) TestTranslatesPlural(c *C) {
+-	// no NG() to avoid adding the test string to snappy-pot
+-	var NGtest = NG
+-	c.Assert(NGtest("plural_1", "plural_2", 1), Equals, "translated plural_1")
+-}
+-
+-func (s *i18nTestSuite) TestTranslatedMissingLangNoCrash(c *C) {
+-	setLocale("invalid")
+-
+-	// no G() to avoid adding the test string to snappy-pot
+-	var Gtest = G
+-	c.Assert(Gtest("singular"), Equals, "singular")
+-}
+-
+-func (s *i18nTestSuite) TestInvalidTextDomainDir(c *C) {
+-	bindTextDomain("snappy-test", "/random/not/existing/dir")
+-	setLocale("invalid")
+-
+-	// no G() to avoid adding the test string to snappy-pot
+-	var Gtest = G
+-	c.Assert(Gtest("singular"), Equals, "singular")
+-}
+-
+-func (s *i18nTestSuite) TestLangpackResolverFromLangpack(c *C) {
+-	root := c.MkDir()
+-	localeDir := filepath.Join(root, "/usr/share/locale")
+-	err := os.MkdirAll(localeDir, 0755)
+-	c.Assert(err, IsNil)
+-
+-	d := filepath.Join(root, "/usr/share/locale-langpack")
+-	makeMockTranslations(c, d)
+-	bindTextDomain("snappy-test", localeDir)
+-	setLocale("")
+-
+-	// no G() to avoid adding the test string to snappy-pot
+-	var Gtest = G
+-	c.Assert(Gtest("singular"), Equals, "translated singular", Commentf("test with %q failed", d))
+-}
+-
+-func (s *i18nTestSuite) TestLangpackResolverFromCore(c *C) {
+-	origSnapMountDir := dirs.SnapMountDir
+-	defer func() { dirs.SnapMountDir = origSnapMountDir }()
+-	dirs.SnapMountDir = c.MkDir()
+-
+-	d := filepath.Join(dirs.SnapMountDir, "/core/current/usr/share/locale")
+-	makeMockTranslations(c, d)
+-	bindTextDomain("snappy-test", "/usr/share/locale")
+-	setLocale("")
+-
+-	// no G() to avoid adding the test string to snappy-pot
+-	var Gtest = G
+-	c.Assert(Gtest("singular"), Equals, "translated singular", Commentf("test with %q failed", d))
+-}
+Index: snapd.mvo/tests/main/i18n/task.yaml
+===================================================================
+--- snapd.mvo.orig/tests/main/i18n/task.yaml
++++ snapd.mvo/tests/main/i18n/task.yaml
+@@ -1,6 +1,8 @@
+ summary: Test that i18n works
+ 
+ execute: |
++    # disabled on debian
++    exit 0
+     # The snapd deb from the archive does not contain .mo files, those
+     # are stripped out by the langpack buildd stuff and put into the
+     # the various langpacks.
+Index: snapd.mvo/i18n/i18n.go
+===================================================================
+--- snapd.mvo.orig/i18n/i18n.go
++++ snapd.mvo/i18n/i18n.go
+@@ -21,83 +21,9 @@ package i18n
+ 
+ //go:generate update-pot
+ 
+-import (
```

Comment:
Maybe in the long run we could use build tags to provide a noop implementation at build time?

Reply:
We discussed a way to use go generate for that. Build tags are another thing.

But in fairness, what we need to do is just package that one library and be done with it. We just don't want to block on that with 2.37.